verifier/src/lib.rs

```diff
@@ -202,82 +308,37 @@ pub trait Verifier<T>: Debug {
     }
 }
 
-/// An error that occurs during an `and` operation.
+/// A result message for an `and` operation.
 #[derive(Debug, Clone)]
-pub struct AndError<L, R> {
+pub struct AndResultMessage<L, R> {
     left: VerificationResult<L>,
     right: VerificationResult<R>,
 }
 
-impl<L, R> AndError<L, R> {
+impl<L, R> AndResultMessage<L, R> {
     /// Create a new instance
     pub fn new(left: VerificationResult<L>, right: VerificationResult<R>) -> Self {
         Self { left, right }
     }
 }
 
-/// Turn a `Choice` into a markdown checkbox
-///
-/// Will return one of:
-/// ```raw
-///     - [ ]
-///     - [x]
-/// ```
-fn choice_to_checkbox(choice: Choice) -> &'static str {
-    if bool::from(choice) {
-        "- [x]"
-    } else {
-        "- [ ]"
+impl<L: ResultMessage, R: ResultMessage> ResultMessage for AndResultMessage<L, R> {
+    fn fmt_success(&self, f: &mut Formatter, pad: usize) -> core::fmt::Result {
+        let status = SUCCESS_MESSAGE_INDICATOR;
+        writeln!(f, "{:pad$}{status} Both of the following are true:", "")?;
+        let pad = pad + MESSAGE_INDENT;
+        self.left.fmt_padded(f, pad)?;
+        writeln!(f)?;
+        self.right.fmt_padded(f, pad)
```

Comment:
These lines are repeated 4 times between `AndResultMessage` and `OrResultMessage`. I made an attempt at using  a re-usable method, but it didn't feel like it made it much better.